### DISCUSSION TITLE
SPX - Path type filters, storage redeclare/detach & stop command for lotus worker - Testing & Call Outs

### DISCUSSION THREAD

**Discussion by TippyFlitsUK** [filecoin-project/lotus#9178] [Show and tell]
![banner](https://user-images.githubusercontent.com/63351350/170097020-c19a4cc9-0845-44a4-bf49-002e9fdd6ebe.png)

## Hello SPX Fellows

&nbsp;
Lotus [1.17.1-rc3](https://github.com/filecoin-project/lotus/releases/tag/v1.17.1-rc3) features a trio of new Lotus features/enhancements that we would sincerely appreciate your help in testing.

This assignment is applicable to both mainnet and calibration instances.

Please reach out to @jennijuju or @TippyFlits on Slack if you have any questions! :pray:
</br>
## Task 1: Path type filters
SPs are now able to filter for sector types that are allowed/denied in each storage path! We know that this will be popular with SPs who have been waiting for a way to separate sealed and unsealed sector data. Please refer to the [new docs section](https://lotus.filecoin.io/storage-providers/operate/custom-storage-layout/) for full details and complete the reply card below with your thoughts and observations.

## Task 2: Storage redeclare/detach
The ability to detach and redeclare a storage path has also been a much-requested feature and is now available in [1.17.1-rc3](https://github.com/filecoin-project/lotus/releases/tag/v1.17.1-rc3!

- The new `lotus-miner storage detach` does exactly what it says on the tin and is [featured here](https://lotus.filecoin.io/storage-providers/operate/custom-storage-layout/#detach-storage-paths) in the Lotus docs.
- You can now update sector locations with the `lotus-miner storage redeclare` command if you want to move sector data from one storage path to another storage path. You are all very used to managing and securing your sector data but please review the new [Updating Locations](https://lotus.filecoin.io/storage-providers/operate/custom-storage-layout/#updating-locations) section of the docs for the full recommended procedure.

## Task 3: Stop command for lotus worker 😮 
The ultimate "better late than never" feature has finally arrived!! :wink: You can now use `lotus-worker stop` to stop any worker without getting bombarded by error messages in your lotus-miner logs! 🍾 

## Step 2: Reply card
Please provide the information below after completion of the above steps and tasks:

- [ ] - `AllowTypes` in your `sectorstorage.json` is respected and applied correctly
- [ ] - `DenyTypes` in your `sectorstorage.json` is also respected and applied correctly
- [ ] - I have successfully detached a storage path
- [ ] - I have successfully redeclared a storage path
- [ ] - I have successfully stopped a running Lotus worker and my logs didn't go bananas! 🍌 🍌 
- [ ] - All normal mining tasks appear to be processing as expected
- [ ] - All normal daemon tasks appear to be processing as expected

## Reminders and Risk Mitigations

- Please review the new [Updating Locations](https://lotus.filecoin.io/storage-providers/operate/custom-storage-layout/#updating-locations) section of the docs before tackling the `lotus-miner storage redeclare` task.
- There is not too much that can go terribly wrong with the other tasks :pray:. If you do have any issues, you're probably already on top of it but do let us know by reporting an issue or leaving a comment/question in this thread in the usual way 👇. We'll be monitoring this discussion and will check in regularly.

Issues can be reported at https://github.com/filecoin-project/lotus/issues. Issues should be submitted with:

- Daemon logs 
- GO routine dumps `lotus pprof goroutines` and `lotus-miner pprof goroutines`
- The operation/step at which the problem occurred
- The stack trace or logs as relevant. As always, for large log files, you may upload them to this [Google Drive folder](https://drive.google.com/drive/folders/1oF1Bfej4XxRv9YZDJTf2EbdctoWGKuQW) and link them in the issue.

**Comment by stuberman** [comment]
I do not have additional storage paths on my miner, so I doubt I can really help on that testing.
I can test the stop functionality.

**Reply by TippyFlitsUK** [reply]
Of course, no problem at all!! Thanks Stu!! 🙏

**Comment by stuberman** [comment]
- [x]  - I have successfully stopped a running Lotus worker and my logs didn't go bananas! 🍌 🍌
 - [x] - All normal mining tasks appear to be processing as expected
 - [x] - All normal daemon tasks appear to be processing as expected

**Reply by stuberman** [reply]
Running `lotus-worker stop` while lotus-miner is not running produces this and lotus-worker does not stop

> 2022-08-19T00:01:30.847Z	WARN	main	lotus-worker/main.go:115	cannot dial address ws://192.168.1.96:3456/rpc/v0 for websocket: bad handshake:
>     github.com/filecoin-project/go-jsonrpc.websocketClient.func1
>         /home/stuart/go/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.1.5/client.go:193
>   - websocket: bad handshake
>

**Reply by TippyFlitsUK** [reply]
Thanks Stu, I get back to you on this. 🙏

**Comment by cryptowhizzard** [comment]
I am running this on f0707721

- [x] - `AllowTypes` in your `sectorstorage.json` is respected and applied correctly
- [x] - `DenyTypes` in your `sectorstorage.json` is also respected and applied correctly
- [X] - I have successfully detached a storage path
- [X] - I have successfully redeclared a storage path
- [X] - I have successfully stopped a running Lotus worker and my logs didn't go bananas! 🍌 🍌 
- [x] - All normal mining tasks appear to be processing as expected
- [x] - All normal daemon tasks appear to be processing as expected

I ran into trouble yesterday ( The chain does not want to sync again after it lost sync ). I am working on it to resolve.

Edit 23-8 . I discovered that when the DNS resolving is done on the machine the chain will go out of sync and stay (slightly) out of sync. This caused the problems. All is running fine now.

**Comment by namoo1** [comment]
# Meest NAM
```
/dev/md126      3.6T  1.5T  2.0T  42% /storage
/dev/sdc1       131T   24K  124T   1% /storage1

```
Added new storage (120T) for "sector" movement
We are going to do a new feature, detach and clare test soon.

In the past, when "retrieve" occurs a lot, "storage" reads HDDs 24 hours a day (because they are unsealed)
Like this upgrade, I'm very happy to announce Unsealed separately and be able to use it for storage for storage.

- [X] - AllowTypes in your sectorstorage.json is respected and applied correctly
- [X] - DenyTypes in your sectorstorage.json is also respected and applied correctly
- [X] - I have successfully detached a storage path
- [X] - I have successfully redeclared a storage path
- [X] - I have successfully stopped a running Lotus worker and my logs didn't go bananas! 🍌 🍌
- [X] - All normal mining tasks appear to be processing as expected
- [X] - All normal daemon tasks appear to be processing as expected

`lotus version 1.17.1-rc3+calibnet+git.079dab0d7`

**Comment by SBudo** [comment]
Mainnet

- [ ] AllowTypes in your sectorstorage.json is respected and applied correctly
- [ ] DenyTypes in your sectorstorage.json is also respected and applied correctly
- [x] I have successfully detached a storage path
- [x] I have successfully redeclared a storage path
- [x] I have successfully stopped a running Lotus worker and my logs didn't go bananas! 🍌 🍌
- [x] All normal mining tasks appear to be processing as expected
- [x] All normal daemon tasks appear to be processing as expected

**Comment by tmyuu** [comment]
## Step 2: Reply card
Please provide the information below after completion of the above steps and tasks:

* [ ]  - `AllowTypes` in your `sectorstorage.json` is respected and applied correctly
* [ ]  - `DenyTypes` in your `sectorstorage.json` is also respected and applied correctly
* [ ]  - I have successfully detached a storage path
* [ ]  - I have successfully redeclared a storage path
* [ ]  - I have successfully stopped a running Lotus worker and my logs didn't go bananas! 🍌 🍌
* [ ]  - All normal mining tasks appear to be processing as expected
* [ ]  - All normal daemon tasks appear to be processing as expected
